src/rust-crypto/rust-crypto.ts

```diff
@@ -1472,6 +1473,30 @@ export class RustCrypto extends TypedEventEmitter<RustCryptoEvents, CryptoEventH
         return batch;
     }
 
+    /**
+     * Implementation of {@link CryptoApi#resetEncryption}.
+     */
+    public async resetEncryption(authUploadDeviceSigningKeys: UIAuthCallback<void>): Promise<void> {
+        const backupEnabled = (await this.backupManager.getActiveBackupVersion()) !== null;
+
+        // Disable backup, and delete all the backups from the server
+        await this.backupManager.deleteAllKeyBackupVersions();
+
+        // Disable the recovery key and the secret storage
+        await this.secretStorage.setDefaultKeyId(null);
+
+        // Reset the cross-signing keys
+        await this.crossSigningIdentity.bootstrapCrossSigning({
+            setupNewCrossSigning: true,
+            authUploadDeviceSigningKeys,
+        });
+
+        // If key backup was enabled, we create a new backup
```

Comment:
Why only if backup was enabled?
That means that if there was a backup that was not working (not trusted signed correctly for example), then the `resetEncryption` flow won't put me back in a working state?

Maybe what we want to check here is if there was a backup server side, or even better do it unless the opt-out flag is set (account data `m.org.matrix.custom.backup_disabled`)?

Reply:
I think that the `rust-sdk` is ultimatly looking at the opt-out flag `m.org.matrix.custom.backup_disabled`, but there is also a local setting to force disable

https://github.com/matrix-org/matrix-rust-sdk/blob/a528624274ecd557427835511eb24513f00696d3/crates/matrix-sdk/src/encryption/recovery/mod.rs#L500-L508

So I believe this code is not consistent with rust. 
Even if the rust doc is misleading https://docs.rs/matrix-sdk/latest/matrix_sdk/encryption/recovery/struct.Recovery.html#method.reset_identity
> Finally, re-enable key backups (only if they were already enabled)

It is unfortunatly not the same `enabled` meaning

Reply:
As discussed elsewhere: ugh, yes, this is not what Element X is doing.

I think we should just enable backups here, regardless of whether they were enabled before. If you're resetting, you're going to be looking at the settings dialog, so can see if you're being opted in to keybackup against your will;  and I'd rather err on the side of just making it work for everyone.